src/core/lib/gprpp/thd_posix.cc

```diff
@@ -48,6 +49,23 @@ struct thd_arg {
   bool tracked;
 };
 
+size_t RoundUpToPageSize(size_t size) {
+  size_t page_size = static_cast<size_t>(sysconf(_SC_PAGESIZE));
```

Comment:
static size_t page_size = ...

As I said in the previous comment, there is no need to read this value every time RoundUpToPageSize is called.